src/dispatching/dialogue/mod.rs

```diff
@@ -179,3 +171,130 @@ where
         self.storage.clone().remove_dialogue(self.chat_id).await
     }
 }
+
+/// Perform a dialogue FSM transition.
+///
+/// This macro expands to a [`dptree::Handler`] that filters your dialogue
+/// state: if the state enumeration is of a certain variant, the execution
+/// continues; otherwise, `dptree` will try the next branch.
+///
+/// Variants can take the following forms:
+///
+///  - `State::MyVariant` for empty variants;
+///  - `State::MyVariant(param1, ..., paramN)` for function-like variants;
+///  - `State::MyVariant { param1, ..., paramN }` for `struct`-like variants.
+///
+/// In the first case, this macro results in a simple [`dptree::filter`]; in the
+/// second and third cases, this macro results in [`dptree::filter_map`] that
+/// passes the payload of `MyVariant` to the next handler if the match occurs.
+/// (This next handler can be an endpoint or a more complex one.) The payload
+/// format depend on the form of `MyVariant`:
+///
+///  - For `State::MyVariant(param)`, the payload is `param`.
+///  - For `State::MyVariant(param1, ..., paramN)`, the payload is `(param1,
+///    ..., paramN)` (where `N` > 1).
+///  - For `State::MyVariant { param1, ..., paramN }`, the payload is `(param1,
+///    ..., paramN)`.
+///
+/// ## Dependency requirements
+///
+///  - Your dialogue state enumeration `State`.
+#[macro_export]
+macro_rules! handler {
+    ($($variant:ident)::*) => {
+        dptree::filter(|state| matches!(state, $($variant)::*))
```

Comment:
This should use `$crate::dptree::...`, otherwise there may be problems if `dptree` doesn't point to the `dptree` _crate_ in a given context.

Reply:
Fixed.